var/spack/repos/builtin/packages/py-pint-xarray/package.py

```diff
@@ -10,18 +10,35 @@ class PyPintXarray(PythonPackage):
     """A convenience wrapper for using pint with xarray"""
 
     homepage = "https://github.com/xarray-contrib/pint-xarray"
-    pypi = "pint-xarray/pint-xarray-0.2.1.tar.gz"
+    pypi = "pint-xarray/pint_xarray-0.4.tar.gz"
 
     license("Apache-2.0")
 
+    version("0.4", sha256="b6b737a9c46dfb14a8598c27a71100496994c9d79dab61fd77f0d2685ae7065e")
     version("0.3", sha256="3545dfa78bee3f98eba29b8bd17500e3b5cb7c7b03a2c2781c4d4d59b6a82841")
     version("0.2.1", sha256="1ee6bf74ee7b52b946f226a96469276fa4f5c68f7381c1b2aae66852562cb275")
 
-    depends_on("py-setuptools@42:", type="build")
-    depends_on("py-setuptools-scm@3.4:+toml", type="build")
-
-    depends_on("python@3.8:", when="@0.3:", type=("build", "run"))
-    depends_on("py-numpy@1.17:", type=("build", "run"))
-    depends_on("py-xarray@0.16.1:", type=("build", "run"))
-    depends_on("py-pint@0.16:", type=("build", "run"))
-    depends_on("py-importlib-metadata", when="@0.2.1 ^python@:3.7", type=("build", "run"))
+    with when("@0.4:"):
+        depends_on("py-setuptools@64:", type="build")
+        depends_on("py-setuptools-scm@7.0:+toml", type="build")
+        depends_on("python@3.9:", when="@0.4:", type=("build", "run"))
+        depends_on("py-numpy@1.23:", type=("build", "run"))
+        depends_on("py-xarray@2022.06.0:", type=("build", "run"))
+        depends_on("py-pint@0.21:", type=("build", "run"))
+
+    with when("@:0.3"):
```

Comment:
fine with the :0.3 but in principle it can be dropped since it's all lowerbounds that are bumped in 0.4:

Reply:
The more strict clause or version dependent clause will always override, correct?